extensions/formats/stanag4676/format/src/main/java/mil/nga/giat/geowave/format/stanag4676/image/ImageChipDataAdapter.java

```diff
@@ -112,4 +113,43 @@ public void fromBinary(
 		return null;
 	}
 
+	@Override
+	public int getPositionOfOrderedField(
+			final CommonIndexModel model,
+			final ByteArrayId fieldId ) {
+		int i = 0;
+		for (final NumericDimensionField<? extends CommonIndexValue> dimensionField : model.getDimensions()) {
+			if (fieldId.equals(dimensionField.getFieldId())) {
+				return i;
+			}
+			i++;
+		}
+		if (fieldId.equals(IMAGE_FIELD_ID)) {
```

Comment:
I know you have something like this everywhere ... what's going on exactly?  I see this 'fieldId.equals(CONSTANT)' in all the adapters but it doesn't make obvious sense to me immediately...